compiler/qsc/src/interpret.rs

```diff
@@ -315,6 +318,72 @@ impl Interpreter {
         })
     }
 
+    /// Given a package ID, returns all the global items in the package.
+    /// Note this does not currently include re-exports.
+    pub fn get_global_items(
+        &self,
+        package_id: PackageId,
+    ) -> Vec<(Vec<Rc<str>>, Rc<str>, fir::StoreItemId)> {
```

Comment:
I think you are right, that will probably be cleaner. I'll update the API and keep `fir::StoreItemId` as an internal thing in favor of `Value` that's already an interop concept. I also like your naming suggestion you made in the other chat, so I'll use `user_globals()` and `source_globals()` is very clean!

Reply:
I made the change, and it works but honestly feels a little awkward. Because I can't return a `Value` directly across into Python and `ValueWrapper` doesn't support round-tripping `GlobalCallable` I end up having to pull out the `fir::StoreItemId` and `FunctorApp` in a conversion method. While I could try to put this conversion logic into `ValueWrapper` so that `Value::Global` gets converted into `GlobalCallable` that just moves the conversion logic and involves another class without really simplifying anything. I pushed up the change, because from the perspective of the interpret.rs file everything looks clean, but now the Rust code that does Python interop handles all of the weird conversions itself and still touches `fir::StoreItemId` so other than APIs in one file I'm not sure this gained us much.

Reply:
I'm surprised the interop ended up being ugly, but I still think it's a worthy tradeoff for keeping the native interface clean. That one gets used by more consumers so it deserves the extra care IMO